sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+}
+
+// @public
+class BatchSetTierRequest extends BatchRequest {
+  constructor();
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  addSubRequest(url: string, credential: SharedKeyCredential | AnonymousCredential | TokenCredential, tier: Models.AccessTier, options?: BlobSetTierOptions): Promise<void>;
+}
+
+// @public (undocumented)
+interface BatchSubRequest {
+  credential: SharedKeyCredential | AnonymousCredential | TokenCredential;
+  url: string;
+}
+
+// @public (undocumented)
+interface BatchSubResponse {
+  _request: BatchSubRequest;
+  bodyAsText?: string;
+  errorCode?: string;
+  headers: HttpHeaders;
+  status: number;
+  statusMessage: string;
+}
+
+// @public
+interface BlobAbortCopyFromURLOptions {
+  abortSignal?: AbortSignalLike;
+  leaseAccessConditions?: Models.LeaseAccessConditions;
+}
+
+// @public
+interface BlobAcquireLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobBreakLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobChangeLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+class BlobClient extends StorageClient {
+  constructor(connectionString: string, containerName: string, blobName: string, options?: NewPipelineOptions);
+  // WARNING: The type "Models.BlobAbortCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobAbortCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  abortCopyFromURL(copyId: string, options?: BlobAbortCopyFromURLOptions): Promise<Models.BlobAbortCopyFromURLResponse>;
+  // WARNING: The type "Models.BlobCreateSnapshotResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobCreateSnapshotResponse" needs to be exported by the package (e.g. added to index.ts)
+  createSnapshot(options?: BlobCreateSnapshotOptions): Promise<Models.BlobCreateSnapshotResponse>;
+  // WARNING: The type "Models.BlobDeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobDeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  delete(options?: BlobDeleteOptions): Promise<Models.BlobDeleteResponse>;
+  // WARNING: The type "Models.BlobDownloadResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobDownloadResponse" needs to be exported by the package (e.g. added to index.ts)
+  download(offset?: number, count?: number, options?: BlobDownloadOptions): Promise<Models.BlobDownloadResponse>;
```

Comment:
BlobDownloadResponse has a property named `blobBody` that I find very unfortunately named. It really feels like it's the thing that gives me easy access to the blob decoded as utf-8 but it doesn't...

BlobDownloadResponse also needs an easy way to get the blob body as a string. It's a lot of work today to subscribe to the stream and accumulate the blob contents as a string.

Reply:
`downloadToBuffer` is better than `downloadToStream` (gets us out of the decoding business).